src/rocksdb.rs

```diff
@@ -39,16 +40,134 @@ pub struct ReadOptions {
     inner: rocksdb_ffi::RocksDBReadOptions,
 }
 
+pub struct Snapshot<'a> {
+    db: &'a RocksDB,
+    inner: rocksdb_ffi::RocksDBSnapshot,
+}
+
+pub struct DBIterator {
+//TODO: should have a reference to DB to enforce scope, but it's trickier than I thought to add
+    inner: rocksdb_ffi::RocksDBIterator,
+    direction: Direction,
+    just_seeked: bool
+}
+
+pub enum Direction {
+  forward, reverse
+}
```

Comment:
A DoubleEndedIterator supports deque-like semantics; however, the rocksdb cursors are designed to scan most efficiently in forward order, and they scan slower in reverse. To support DEI semantics, we'd seek between every key, yielding poor performance.

The solution to that is to create a Snapshot, then create 2 iterators on it, one to go forward; the other, reverse.